okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -538,6 +539,18 @@ public void proxy(RoutingContext ctx) {
           stream.resume();
           return; // ctx already set up
         }
+
+        // handle delegate CORS
+        if (ctx.data().containsKey(MainVerticle.CHECK_DELEGATE_CORS)) {
+          ctx.data().remove(MainVerticle.CHECK_DELEGATE_CORS);
+          if (l.stream().anyMatch(mi -> mi.isHandler() && mi.getRoutingEntry().isDelegateCors())) {
+            ctx.data().put(MainVerticle.DELEGATE_CORS, true);
+            stream.resume();
+            ctx.reroute(ctx.request().path());
+            return;
+          }
+        }
+
         pc.setModList(l);
 
         pc.logRequest(ctx, tenantId);
```

Comment:
Please don't import MainVerticle which also includes ProxyService. Can you move all CORS logic to a CorsHandler class? Which is this is then imported/used by ProxyService and MainVerticle.

Reply:
Done.